src/main/java/com/github/randoapp/api/API.java

```diff
@@ -258,9 +258,11 @@ public static Rando uploadImage(File randoFile, Location location) throws Except
 
             if (response.getStatusLine().getStatusCode() == SC_OK) {
                 JSONObject json = readJSON(response);
-                Rando rando = new Rando();
-                rando.imageURL = json.getString(IMAGE_URL_PARAM);
-                rando.date = new Date(json.getLong(CREATION_PARAM));
+                String randoJson = json.getString(RANDO_PARAM);
+                Rando rando = null;
+                if (randoJson != null) {
```

Comment:
If `randoJson` is null we return `null` for `uploadImage` method. After that we will go to `new UploadTask().onOk` handler and remove rando from upload. I'm not sure that response from server without `json.getString(RANDO_PARAM)` is success and we should delete image from db.